### PR TITLE
Remove reward time restrictions

### LINKED ISSUE #754: Remove reward payout time restrictions
Currently there is a restriction where only one reward payout can be active at a time, and each reward payout is active for 60 days (meaning that only one payout can be initiated every 60 days).

This now appears to be an overly-restrictive restriction, as there are use-cases where reward payouts may need to be initiated more frequently. Given that now reward payouts are secured with the `root` permission, any time restriction at all seems less necessary. We should remove the time restriction and add support for multiple _simultaneous payouts_.

Regarding implementation, one approach would be to track activePayoutTotal and payout.totalClaimed and use that to determine the amount available for any particular payout cycle:

```
// When starting a new payout:
payout.totalPayout = fundingPots[0].balance[token] - activePayouts[token]; // the inactive balance

// When finalizing a payout:
activePayouts[token] -= (payout.totalPayout - payout.totalClaimed) // the unclaimed surplus
```

### PR DESCRIPTION
**Pull request by area**
Closes #754 

Completely deprecates the mapping tracking payouts per-token, and moved to tracking whether each payout is active individually. For bookkeeping purposes, I've used the calculations Daniel suggested in the original issue.

I've not reused the original mapping storage slot for this bookkeeping, but I don't think there's a reason we can't? 

Added a new function to `helpers` to generate the square roots and reputation proofs used in claiming payouts, which has removed a lot of duplicate code in the test file.

### CODE REVIEW

File: test/contracts-network/token-locking.js [hunk 1]
```diff
@@ -181,6 +181,8 @@ contract("Token Locking", addresses => {
     it("should correctly increment total lock count", async () => {
       await token.approve(tokenLocking.address, usersTokens, { from: userAddress });
       await tokenLocking.deposit(token.address, usersTokens, { from: userAddress });
+      await fundColonyWithTokens(colony, otherToken);
+      await colony.moveFundsBetweenPots(1, 0, 0, 1, 0, 100, otherToken.address);
```
Comment (kronosapiens) [inline test/contracts-network/token-locking.js]: Why are these lines needed now and not before?

File: contracts/ColonyFunding.sol [hunk 2]
```diff
@@ -276,8 +276,9 @@ contract ColonyFunding is ColonyStorage, PatriciaTreeProofs { // ignore-swc-123
   {
     ITokenLocking tokenLocking = ITokenLocking(IColonyNetwork(colonyNetworkAddress).getTokenLocking());
     uint256 totalLockCount = tokenLocking.lockToken(token);
-
-    require(activeRewardPayouts[_token] == 0, "colony-reward-payout-token-active");
+    uint256 thisPayoutAmount = sub(fundingPots[0].balance[_token], possiblyPendingRewardPayments[_token]);
```
[the remark this answers is not in the document]
Comment (area) [reply contracts/ColonyFunding.sol]: They're not all pending, was my reasoning, at the very least because payouts that people have waived their right to (and so are not pending being claimed) are included.... But maybe this was too conservative on my part. Will change.

File: contracts/ColonyStorage.sol [hunk 3]
```diff
@@ -83,6 +85,8 @@ contract ColonyStorage is CommonStorage, ColonyDataTypes, ColonyNetworkDataTypes
   mapping (uint256 => mapping (uint256 => ExpenditureSlot)) expenditureSlots; // Storage slot 26
   mapping (uint256 => mapping (uint256 => mapping (address => uint256))) expenditureSlotPayouts; // Storage slot 27
 
+  mapping (address => uint256) possiblyPendingRewardPayments; // Storage slot 28
```
[the remark this answers is not in the document]
Comment (area) [reply contracts/ColonyStorage.sol]: I guess we should take these opportunities when we can.

File: test/contracts-network/colony-reward-payouts.js [hunk 4]
```diff
@@ -276,17 +280,98 @@ contract("Colony Reward Payouts", accounts => {
...
+      await forwardTime(SECONDS_PER_DAY * 60 + 1, this);
+
+      await colony.moveFundsBetweenPots(1, 0, 0, 1, 0, 100, otherToken.address);
+      await colony.finalizeRewardPayout(payoutId1);
+
+      const tx2 = await colony.startNextRewardPayout(otherToken.address, ...colonyWideReputationProof);
+      const payoutId2 = tx2.logs[0].args.rewardPayoutId;
+
+      const payout2 = await colony.getRewardPayoutInfo(payoutId2);
+      assert.equal(
```
[the remark this answers is not in the document]
Comment (area) [reply test/contracts-network/colony-reward-payouts.js]: If we just disabled `eslint` where we individually didn't agree with it, we'd end up in a right pickle 😄

File: test/contracts-network/token-locking.js (hunk 1 above)
Comment (area) [reply test/contracts-network/token-locking.js]: I no longer allow a payout to start if there are no tokens to be paid out.

File: test/contracts-network/colony-reward-payouts.js [hunk 5]
```diff
@@ -276,17 +280,98 @@ contract("Colony Reward Payouts", accounts => {
...
+      await colony.moveFundsBetweenPots(1, 0, 0, 1, 0, 100, otherToken.address);
+
+      const tx1 = await colony.startNextRewardPayout(otherToken.address, ...colonyWideReputationProof);
+      const payoutId1 = tx1.logs[0].args.rewardPayoutId;
+
+      await colony.moveFundsBetweenPots(1, 0, 0, 1, 0, 100, otherToken.address);
+
+      const tx2 = await colony.startNextRewardPayout(otherToken.address, ...colonyWideReputationProof);
+      const payoutId2 = tx2.logs[0].args.rewardPayoutId;
+      const { squareRoots } = await getRewardClaimSquareRootsAndProofs(client, tokenLocking, colony, payoutId2, userAddress1);
```
[the remark this answers is not in the document]
Comment (area) [reply test/contracts-network/colony-reward-payouts.js]: The tokens in the pot has, though - some are provided in the `beforeEach`. I guess I've argued I should remove the first `moveFundsBetweenPots` for clarity...

File: test/contracts-network/colony-reward-payouts.js [hunk 6]
```diff
@@ -276,17 +280,98 @@ contract("Colony Reward Payouts", accounts => {
...
+      const payoutId2 = tx2.logs[0].args.rewardPayoutId;
+
+      const payout2 = await colony.getRewardPayoutInfo(payoutId2);
+      assert.equal(
+        payout2.amount,
+        initialFunding
+          .divn(100)
+          .addn(100)
+          .toString(),
+        "Payout has unexpected amount"
```
[the remark this answers is not in the document]
Comment (area) [reply test/contracts-network/colony-reward-payouts.js]: They make the results during a failure much more understandable, so I'm inclined to leave them in if you don't feel strongly.

File: test/contracts-network/colony-reward-payouts.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (kronosapiens) [reply test/contracts-network/colony-reward-payouts.js]: Fair, but in particular `eslint` is myopic when it comes to putting statements on multiple lines.

File: test/contracts-network/colony-reward-payouts.js [hunk 7]
```diff
@@ -276,17 +280,98 @@ contract("Colony Reward Payouts", accounts => {
...
       );
     });
 
-    it("should not be able to create parallel payouts of the same token", async () => {
+    it("should not be able to create parallel payouts of the same token if there are no more tokens to pay out", async () => {
+      await colony.startNextRewardPayout(otherToken.address, ...colonyWideReputationProof);
+      await checkErrorRevert(colony.startNextRewardPayout(otherToken.address, ...colonyWideReputationProof), "colony-reward-payout-no-rewards");
+    });
+
+    it("should be able to create parallel payouts of the same token", async () => {
```
[the remark this answers is not in the document]
Comment (kronosapiens) [reply test/contracts-network/colony-reward-payouts.js]: After more discussion, we decided that this restriction wouldn't actually provide any guarantees since a malicious user could simply make more colonies to get around the restriction.
